src/widget/number-input/number-input.js

```diff
@@ -1,93 +1,173 @@
-import { t } from '../../js/fake-translator';
 import events from '../../js/event';
-import inputModule from '../../js/input';
+import { t } from '../../js/fake-translator';
 import Widget from '../../js/widget';
 
+/** @type {Set<NumberInput>} */
+let numberInputInstances = new Set();
+
+/** @type {WeakMap<HTMLInputElement, HTMLElement>} */
+let questionsByInput = new WeakMap();
```

Comment:
Having this map here feels weird to me but @eyelidlessness has confirmed that the query to get the input container (`.question`) is quite expensive and ends up being run all the time. This especially affects forms with number inputs in repeats.